backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/window/model/lookup/LookupDataSourceContext.java

```diff
@@ -722,5 +723,21 @@ private Object findContextValueOrNull(@NonNull final CtxName variableName)
 			// Value not found
 			return null;
 		}
+
+		@Nullable
+		private String getContextTableName()
+		{
+			try
+			{
+				collectContextValue(PARAM_ContextTableName, false);
+				final Object contextTableNameObj = valuesCollected.get(PARAM_ContextTableName.getName());
+				return contextTableNameObj != null ? contextTableNameObj.toString() : null;
+			}
+			catch (final Exception e)
+			{
+				return null;
```

Comment:
1. why try/catch? what exception are u actually getting? IMHO u don't need to add that guard
2. if u really have to do it, then pls log.warning that exception, don't simply swallow it. If the exception is happening very often and the console is spammed then it means that your code shall do some "if"s and avoid provoking the exception

Reply:
the example where i got IllegalStateException was when i tried to set the Country of a BP. The List wouldn't load and throw the error

Reply:
<details>
<summary>exception</summary>

```
java.lang.IllegalStateException: No TableName defined for DocumentEntityDescriptor{fields.count=12}
	at de.metas.ui.web.window.descriptor.DocumentEntityDescriptor.lambda$getTableName$3(DocumentEntityDescriptor.java:386)
	at java.util.Optional.orElseThrow(Optional.java:290)
	at de.metas.ui.web.window.descriptor.DocumentEntityDescriptor.getTableName(DocumentEntityDescriptor.java:386)
	at de.metas.ui.web.window.model.DocumentEvaluatee.get_ValueIfExists(DocumentEvaluatee.java:235)
	at de.metas.ui.web.window.model.lookup.LookupDataSourceContext$Builder.findContextValueOrNull(LookupDataSourceContext.java:710)
	at de.metas.ui.web.window.model.lookup.LookupDataSourceContext$Builder.collectContextValue(LookupDataSourceContext.java:671)
	at de.metas.ui.web.window.model.lookup.LookupDataSourceContext$Builder.getContextTableName(LookupDataSourceContext.java:732)
	at de.metas.ui.web.window.model.lookup.LookupDataSourceContext$Builder.build(LookupDataSourceContext.java:478)
	at de.metas.ui.web.window.model.lookup.LookupDataSourceAdapter.findEntities(LookupDataSourceAdapter.java:97)
	at de.metas.ui.web.window.model.lookup.LookupDataSource.findEntities(LookupDataSource.java:49)
	at de.metas.ui.web.window.model.DocumentField.getLookupValuesForQuery(DocumentField.java:428)
	at de.metas.ui.web.window.model.Document.getFieldLookupValuesForQuery(Document.java:1608)
	at de.metas.ui.web.address.AddressRestController.getAttributeTypeahead(AddressRestController.java:138)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:197)
	at org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:141)
	at org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:106)
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandlerMethod(RequestMappingHandlerAdapter.java:894)
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:808)
	at org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:87)
	at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:1060)
	at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:962)
	at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:1006)
	at org.springframework.web.servlet.FrameworkServlet.doGet(FrameworkServlet.java:898)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:626)
	at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:883)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:733)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:227)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162)
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:53)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162)
	at de.metas.ui.web.config.WebConfig$1.doFilter(WebConfig.java:69)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162)
	at de.metas.ui.web.CORSFilter.doFilter(CORSFilter.java:45)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162)
	at de.metas.ui.web.config.ServletLoggingFilter.doFilter(ServletLoggingFilter.java:91)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162)
	at org.springframework.session.web.http.SessionRepositoryFilter.doFilterInternal(SessionRepositoryFilter.java:141)
	at org.springframework.session.web.http.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:82)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162)
	at org.springframework.boot.actuate.metrics.web.servlet.WebMvcMetricsFilter.doFilterInternal(WebMvcMetricsFilter.java:93)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162)
	at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:201)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162)
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:202)
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:97)
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:542)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:143)
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92)
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:78)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:346)
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:374)
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:65)
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:887)
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1684)
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	at java.lang.Thread.run(Thread.java:750)
```

</details>